dist/distribution_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dist
+
+import (
+	"math"
+	"sort"
+	"testing"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/stat"
+)
+
+// fullDist is a distribution that implements all the basic functions.
+type fullDist interface {
+	CDF(x float64) float64      // check
+	Entropy() float64           // check
+	ExKurtosis() float64        // check
+	LogProb(x float64) float64  // check
+	Mean() float64              // check
+	Median() float64            // check
+	Prob(x float64) float64     // check
+	Quantile(p float64) float64 // check
+	Rand() float64              // check
+	Skewness() float64          // check
+	StdDev() float64            // check
+	Survival(x float64) float64
+	Variance() float64 // check
+}
+
+// testFullDist tests all of the functions of a fullDist.
+func testFullDist(t *testing.T, f fullDist, i int) {
+	tol := 1e-1
+	n := 1000000
```

Comment:
> On Jun 2, 2015, at 5:15 PM, Dan Kortschak <notifications@github.com> wrote:
> 
> In dist/distribution_test.go <https://github.com/gonum/stat/pull/42#discussion_r31584347>:
> 
> > +	LogProb(x float64) float64  // check
> > +	Mean() float64              // check
> > +	Median() float64            // check
> > +	Prob(x float64) float64     // check
> > +	Quantile(p float64) float64 // check
> > +	Rand() float64              // check
> > +	Skewness() float64          // check
> > +	StdDev() float64            // check
> > +	Survival(x float64) float64
> > +	Variance() float64 // check
> > +}
> > +
> > +// testFullDist tests all of the functions of a fullDist.
> > +func testFullDist(t *testing.T, f fullDist, i int) {
> > +	tol := 1e-1
> > +	n := 1000000
> Yeah, the reasoning is that it's easier to read 1e6 than 1000000, but if you do 1e6 it's declared as float64, so you need to make it untyped or var n int = 1e6 or n := int(1e6), neither of which look that good.
> 
Yea, I worked through all of those logical steps, but hadn’t thought of the simpler “const” option.